proposals/0929-legacy-architecture-removal.md

```diff
@@ -0,0 +1,347 @@
+
+While we [already implemented](https://github.com/reactwg/react-native-new-architecture/discussions/290) part of this RFC earlier this year, we want to share it with the community and partners early on to give a heads up on what’s coming next for React Native in the coming releases.
+
+## Timeline
+
+The following is a rough timeline of the changes related to this RFC:
+
+![Diagram showing the timeline for Legacy Arch removal](assets/0929-legacyarch-removal-timeline.png)
+
+### Already implemented steps
+
+* **2024-05-16:** New Architecture [is considered beta](https://github.com/reactwg/react-native-new-architecture/discussions/189)
+* **2024-10-23: [React Native 0.76]** New Architecture is considered **[stable and generally available](https://reactnative.dev/blog/2024/10/23/the-new-architecture-is-here)**
+* **2025-06-02:** Legacy Architecture [is considered frozen](https://github.com/reactwg/react-native-new-architecture/discussions/290). Specifically from that day:
+    * We stopped working on Legacy Architecture.
+    * We stopped accepting PRs that modify the Legacy Architecture.
+    * We stopped testing the Legacy Architecture when releasing React Native.
+* **2025-06-12: [React Native 0.80]** New Architecture [warnings are added](https://reactnative.dev/blog/2025/06/12/react-native-0.80#legacy-architecture-freezing--warnings)
+    * Users will see a warning on DevTools when using Legacy Architecture.
+    * Moreover, specific warnings have been added to notify the user if they’re using a pattern that won’t be supported by interop layers
+* **2025-08-12: [React Native 0.81]** Further New Architecture classes are marked as `@Deprecated`
+    * Users will be notified at build time if using a `@Deprecated` class through a build warning.
+
+### Upcoming steps
+
+* **2025-10-x: [React Native 0.82]** New Architecture opt-out is removed.
+    * Users won’t be able to opt out of the newArchitecture anymore.
+    * Specifically:
+        * On Android, setting `newArchEnabled=false` will result in a warning message and the property will be hardcoded to `true`
+        * To keep compatibility with libraries, we will set `newArchEnabled=true` if a user removes the `newArchEnabled` line from their `gradle.properties`.
+        * On iOS
+            * Installing pods with ``RCT_NEW_ARCH_ENABLED=0`` will result in a warning message and the property will be hardcoded to `1`
+            * Changing the `RCTNewArchEnabled` flag in the `Info.plist` will be ignored
+            * Overriding methods such as `newArchEnabled` in the AppDelegate will be ignored
+        * To keep compatibility with libraries, we will set `RCT_NEW_ARCH_ENABLED=1`in all the podspecs, including the one autolinked. This will give library authors the time to remove the legacy arch code from their library.
+    * We don’t forecast removal of Legacy Architecture APIs in this version (i.e. Legacy Architecture will still be inside React Native core, but not accessible).
```

Comment:
this list misses a point where we explain that we are deprecating legacy architecture classes to give developers signal when their code is using them.

Reply:
There is a section above that reads:

```
* **2025-08-12: [React Native 0.81]** Further New Architecture classes are marked as `@Deprecated`
    * Users will be notified at build time if using a `@Deprecated` class through a build warning.
```

I know that deprecation has been added throughout different versions, not exactly only in 0.81